IoT-Sample/Swift/IoTSampleSwift/ConnectionViewController.swift

```diff
@@ -19,207 +19,273 @@ import AWSMobileClient
-                                {
-                                    DispatchQueue.main.asyncAfter(deadline: .now()+2, execute: {
-                                        self.logTextView.text = "Using certificate: \(certificateId!)"
-                                        self.iotDataManager.connect( withClientId: uuid, cleanSession:true, certificateId:certificateId!, statusCallback: mqttEventCallback)
+    func handleDisconnect() {
+        let tabBarViewController = tabBarController as! IoTSampleTabBarController
+        self.connectButton.isHidden = false
+        self.connectIoTDataWebSocket.isHidden = false
+        activityIndicatorView.startAnimating()
+        logTextView.text = "Disconnecting..."
 
-                                    })
-                                }
-                                return nil
-                            })
-                        }
-                        else
-                        {
-                            DispatchQueue.main.async {
-                                sender.isEnabled = true
-                                self.activityIndicatorView.stopAnimating()
-                                self.logTextView.text = "Unable to create keys and/or certificate, check values in Constants.swift"
-                            }
-                        }
-                    } )
-                }
+        DispatchQueue.global(qos: DispatchQoS.QoSClass.default).async {
+            self.iotDataManager.disconnect();
+            DispatchQueue.main.async {
+                self.connected = false
+                tabBarViewController.viewControllers = [ self, self.configurationViewController ]
             }
-            else
-            {
-                let uuid = UUID().uuidString;
+        }
+    }
 
-                // Connect to the AWS IoT service
-                iotDataManager.connect( withClientId: uuid, cleanSession:true, certificateId:certificateId!, statusCallback: mqttEventCallback)
+    func searchForExistingCertificateIdInBundle() -> String? {
+        let defaults = UserDefaults.standard
+        // No certificate ID has been stored in the user defaults; check to see if any .p12 files
+        // exist in the bundle.
+        let myBundle = Bundle.main
+        let myImages = myBundle.paths(forResourcesOfType: "p12" as String, inDirectory:nil)
+        let uuid = UUID().uuidString
+        
+        if (myImages.count > 0) {
+            // At least one PKCS12 file exists in the bundle.  Attempt to load the first one
+            // into the keychain (the others are ignored), and set the certificate ID in the
+            // user defaults as the filename.  If the PKCS12 file requires a passphrase,
+            // you'll need to provide that here; this code is written to expect that the
+            // PKCS12 file will not have a passphrase.
+            if let data = try? Data(contentsOf: URL(fileURLWithPath: myImages[0])) {
```

Comment:
Let's guard/let this, so we can report if the data isn't correctly decrypted. We'd want customers to handle the error cases with some debugging, so it'd be good to include that in our source even if we don't have any meaningful action we can take in the case of an error other than printing it

Reply:
Again, not touching this code, sorry -- cruft from before can carry over, and we can update at a later time.

Reply:
Updated.